tests/ops/api/v1/endpoints/test_config_endpoints.py

```diff
@@ -539,6 +607,39 @@ def test_reset_application_config_non_existing(
         response_settings = response.json()
         assert response_settings == {}
 
+    def test_reset_removes_all_cors_domain_from_middleware(
+        self,
+        api_client: TestClient,
+        generate_auth_header,
+        url,
+        db: Session,
+        cors_middleware,
+        payload,
+    ):
+        auth_header = generate_auth_header([scopes.CONFIG_UPDATE])
+        response = api_client.patch(
+            url,
+            headers=auth_header,
+            json=payload,
+        )
+        assert response.status_code == 200
+
+        auth_header = generate_auth_header([scopes.CONFIG_UPDATE])
+        response = api_client.delete(
+            url,
+            headers=auth_header,
+        )
+        assert response.status_code == 200
+        # Check the app config direcitly to make sure it's reset and
+        # by proxy check that the CORS domains are reset.
+        # The middleware isn't checked directly because it's
+        # possible that the config set will still have some 
+        # domains and when the `api_set` is reset the middleware
+        # will still have the `config_set` domains loaded in.
```

Comment:
right so this is the behavior that i think would be desirable, but per my above comment i'm worried that this isn't actually exactly what's happening. happy to talk through this more because it gets a bit tricky (and perhaps counterintuitive) with the config proxy, but i actually think that if we're going for this behavior, we can accomplish it with what i suggested above 👍